chainstate/test-suite/src/tests/pos_processing_tests.rs

```diff
@@ -1743,3 +1743,200 @@ fn pos_stake_testnet_genesis(#[case] seed: Seed) {
+    tf.set_time_seconds_since_epoch(parent.block_timestamp().as_int_seconds() + 1);
+
+    let staking_destination = Destination::PublicKey(PublicKey::from_private_key(staker_sk));
+    let kernel_outputs =
+        vec![TxOutput::ProduceBlockFromStake(staking_destination.clone(), pool_id)];
+
+    let kernel_sig = produce_kernel_signature(
+        tf,
+        staker_sk,
+        kernel_outputs.as_slice(),
+        staking_destination,
+        parent.block_id(),
+        kernel_input_outpoint.clone(),
+    );
+
+    let new_block_height = parent.block_height().next_height();
+    let current_difficulty = calculate_new_target(tf, new_block_height).unwrap();
+    let initial_randomness = tf.chainstate.get_chain_config().initial_randomness();
+    let pool_balance =
+        PoSAccountingStorageRead::<TipStorageTag>::get_pool_balance(&tf.storage, pool_id)
+            .unwrap()
+            .unwrap();
+    let (pos_data, block_timestamp) = pos_mine(
+        BlockTimestamp::from_duration_since_epoch(tf.current_time()),
+        kernel_input_outpoint,
+        InputWitness::Standard(kernel_sig),
+        vrf_sk,
+        PoSRandomness::new(initial_randomness),
+        pool_id,
+        pool_balance,
+        0,
+        current_difficulty,
+    )
+    .unwrap();
+    let consensus_data = ConsensusData::PoS(Box::new(pos_data));
+
+    let block = tf
+        .make_block_builder()
+        .with_consensus_data(consensus_data)
+        .with_block_signing_key(staker_sk.clone())
+        .with_timestamp(block_timestamp)
+        .with_transactions(transactions)
+        .with_reward(kernel_outputs)
+        .build();
+
+    tf.process_block(block.clone(), BlockSource::Local).unwrap().unwrap();
+
+    block
+}
+
+#[rstest]
+#[trace]
+#[case(Seed::from_entropy())]
+fn pos_reorg(#[case] seed: Seed) {
+    logging::init_logging::<std::path::PathBuf>(None);
+
+    let mut rng = make_seedable_rng(seed);
+    let upgrades = vec![
+        (
+            BlockHeight::new(0),
+            UpgradeVersion::ConsensusUpgrade(ConsensusUpgrade::IgnoreConsensus),
+        ),
+        (
+            BlockHeight::new(1),
+            UpgradeVersion::ConsensusUpgrade(ConsensusUpgrade::PoS {
+                initial_difficulty: MIN_DIFFICULTY.into(),
+                config: create_unittest_pos_config(),
+            }),
+        ),
+    ];
+    let genesis_pool_id = PoolId::new(H256::zero());
+    let (vrf1_sk, vrf1_pk) = VRFPrivateKey::new_from_rng(&mut rng, VRFKeyKind::Schnorrkel);
+    let (staker1_sk, staker1_pk) = PrivateKey::new_from_rng(&mut rng, KeyKind::Secp256k1Schnorr);
+    let (vrf2_sk, vrf2_pk) = VRFPrivateKey::new_from_rng(&mut rng, VRFKeyKind::Schnorrkel);
+    let (staker2_sk, staker2_pk) = PrivateKey::new_from_rng(&mut rng, KeyKind::Secp256k1Schnorr);
+    let mint_amount = Amount::from_atoms(100_000_000 * common::chain::Mlt::ATOMS_PER_MLT);
+    let stake_amount = Amount::from_atoms(40_000_000 * common::chain::Mlt::ATOMS_PER_MLT);
```

Comment:
Small hint: you could use `ChainConfig::min_stake_pool_pledge()` to avoid manual amounts

Reply:
Right, but the `ChainConfig` instance is not constructed at that moment yet.